### PR TITLE
Fix slicer assignment bug

### LINKED ISSUE #26939: BUG: cannot set Series with reverse slicer
Minimal example:
```
>>> import pandas as pd
>>> s = pd.Series(index=range(2010, 2020))
>>> s.loc[2015:2010:-1] = [6, 5, 4, 3, 2, 1]
Traceback (most recent call last):
[...]
ValueError: cannot set using a slice indexer with a different length than the value
```

I see no reason why this shouldn't work, as setting with the forward slicer works without problems, and *getting* with the reverse slicer also works without issue:
```
>>> # turn list around because slicer is (not) reversed compared to above
>>> s.loc[2010:2015] = [6, 5, 4, 3, 2, 1][::-1]
>>> s
2010    1.0
2011    2.0
2012    3.0
2013    4.0
2014    5.0
2015    6.0
2016    NaN
2017    NaN
2018    NaN
2019    NaN
dtype: float64
>>> s.loc[2015:2010:-1] == [6, 5, 4, 3, 2, 1]  # comparison, not assignment
2015    True
2014    True
2013    True
2012    True
2011    True
2010    True
dtype: bool
```

PS: For the failure, it does not matter if the RHS is a np.array, etc.

### PR DESCRIPTION
**Pull request by RoeiRaz**
- [x] closes #26939 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/indexers.py [hunk 1]
```diff
@@ -226,6 +226,9 @@ def length_of_indexer(indexer, target=None) -> int:
         if step is None:
             step = 1
         elif step < 0:
+            t = stop
+            stop = start + 1
+            start = t + 1
```
Comment (simonjayhawkins) [inline pandas/core/indexers.py]: can you eliminate the temporary variable with `start, stop = stop+1, start+1`

**Review by TomAugspurger — COMMENTED** [review]
Looks good.  Can you add a release note in 1.0.0?

Are there any issues with negative steps other than `-1`, or is the adjustment to start and stop correct for all negative steps?

**Review by TomAugspurger — APPROVED** [review]

**Comment by jbrockmendel** [comment]
LGTM

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Nice PR - thanks @RoeiRaz !

**Pull request merged**